sdcm/utils/common.py

```diff
@@ -3143,3 +3143,8 @@ def SoftTimeoutContext(timeout: int, operation: str):  # pylint: disable=invalid
     if duration > timeout:
         SoftTimeoutEvent(operation=operation, soft_timeout=timeout,
                          duration=duration).publish_or_dump()
+
+
+def raise_exception_in_thread(thread: threading.Thread, exception_type: Type[BaseException]):
+    res = ctypes.pythonapi.PyThreadState_SetAsyncExc(ctypes.c_ulong(thread.ident), ctypes.py_object(exception_type))
```

Comment:
Maybe we should raise if result of the api call is not as expected, like:
```
res = pythonapi.PyThreadState_SetAsyncExc(c_long(thread.ident),
                                                      py_object(exception_type))
        if res == 0:
            raise ValueError("invalid thread id")
        elif res != 1:
            # "if it returns a number greater than one, you're in trouble,
            # and you should call it again with exc=NULL to revert the effect"
            pythonapi.PyThreadState_SetAsyncExc(c_long(tid), None)
            raise SystemError("PyThreadState_SetAsyncExc failed")
```

Reply:
this case might have, if calling this function multiple times without holding the GIL.
I don't think it could in our case.

But I still didn't see it happening as I would expect